lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -1910,26 +1908,72 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
           }
         }
 
-        // Handle cases that axes are explicitly specified.
-        // Extract the axes values from the axes operand.
-        // This really shouldn't happen but it helps pass weird tests.
-        // TODO: Derive the chosen axes from the data type and final result type
-        // instead of using the dynamic axes at operand[1].
         if (!binder.tensorOperandAtIndex(axes, 1)) {
-          Torch::BaseTensorType axesType =
-              axes.getType().cast<Torch::BaseTensorType>();
-          auto sizes = axesType.getSizes();
-          for (int i = 0; i < sizes[0]; i++) {
-            Value selectIndex = rewriter.create<Torch::ConstantIntOp>(
-                binder.getLoc(), rewriter.getI64IntegerAttr(i));
-            Value extract = rewriter.create<Torch::AtenSelectIntOp>(
-                binder.getLoc(),
-                axesType.getWithSizesAndDtype(llvm::SmallVector<int64_t>{1},
-                                              axesType.getOptionalDtype()),
-                axes, zero, selectIndex);
-            Value dim = rewriter.create<Torch::AtenItemOp>(binder.getLoc(),
-                                                           torchIntTy, extract);
-            axesList.push_back(dim);
+          SmallVector<int64_t> inputShape{dataTy.getSizes()};
+          SmallVector<int64_t> resultShape{resultType.getSizes()};
+          // if the shapes are equal, none of the dims is reduced
+          if (llvm::equal(inputShape, resultShape)) {
+            // simply fill in the op and return
+            rewriter.replaceOp(binder.op, data);
+            return success();
+          }
+
+          // Try to derive the static axes from the data type and final result
+          // type. This algorithm won't work if the input shape has dupliacte
+          // dim: [2,2,2] -> [2,2], axes=1/2/3
+          if (areAllElementsDistinct(inputShape)) {
```

Comment:
You still are making assumptions that shapes are well known. It is possible have dynamic dimensions resulting in mismatchings due to weirdness in shape inference.

Reply:
Yes. This is to pass some static know test case.